### ISSUE TITLE
Error message - Could not acquire data - time data error

### ISSUE THREAD

**Issue by waynedn** [MarcoGos/davis_vantage#5]
Installed your integration yesterday for my Davis Vantage Vue with a 6555 IP connection unit.  The integration found my Davis unit but I am getting the following error message in my logs with no data coming back:

Couldn't acquire data from tcp:10.0.8.57:22222: time data '281:61' does not match format '%H:%M'
Couldn't acquire data from tcp:10.0.8.57:22222: [Errno 32] Broken pipe

Any guidance to fix?

**Comment by MarcoGos** [comment]
That's hard to say. It looks like the time part of the high or low values returned from the device is incorrect. What do the high/low values and the corresponding date/time values read bij WeatherLink software say?

**Comment by waynedn** [comment]
My data logger has always communicated directly to the Weatherlink.com website.  I installed to Weatherlink software on one of my Windows machines but I am unable to connect  to the data logger using TCP/IP in the setup.  I can browse to the data logger and get the screen which shows the details for the logger.  It is configured for port 22222.  Can it be configured to upload to weartehrlink.com while I am trying to connect directly on the LAN through the software?

It is as if the data logger is refusing a connection. 

Any ideas?

**Comment by vk2him** [comment]
I'm getting a number of errors in my logs connecting via a networked Vantage Pro Plus. It's been working fine for years talking to Weatherdisplay via network. Heres the errors

````
2024-03-06 08:21:19.925 WARNING (MainThread) [homeassistant.util.async_] Detected blocking call to sleep inside the event loop by custom integration 'davis_vantage' at custom_components/davis_vantage/client.py, line 252: data["ArchiveInterval"] = self._vantagepro2.archive_period, please create a bug report at https://github.com/MarcoGos/davis_vantage/issues
2024-03-06 08:21:21.701 WARNING (MainThread) [custom_components.davis_vantage] Couldn't acquire data from tcp:192.168.1.61:22222: string argument without an encoding
2024-03-06 08:22:35.707 WARNING (MainThread) [custom_components.davis_vantage] Couldn't acquire data from tcp:192.168.1.61:22222: [Errno 32] Broken pipe
2024-03-06 08:26:02.756 WARNING (MainThread) [custom_components.davis_vantage] Couldn't acquire data from tcp:192.168.1.61:22222: No valid acknowledgement.

**Comment by MarcoGos** [comment]
@vk2him Copied the first issue you've found to #6 and fixed that. The other warnings are common. The data logger doesn't communicate that well. You don't see this is the original software because those warnings are ignored.

**Comment by MarcoGos** [comment]
@waynedn It should be possible to connect the WeatherLink software to the ip logger via local LAN. I don't own a IP logger nor a Vantage Vue, so I cannot try it myself. Also see https://www.weatherlink.com/static/software/weatherlink_software_getting_started_guide_pc_2018_11_28.pdf

**Comment by vk2him** [comment]
Thanks - #6 fixed my issue, the integration is now displaying data.

There is a problem still however - do you want me to log a new issue of can I add it here?

The rain data is not working properly - it's currently raining here at the moment and the integrations isn't recording the current rain data correctly for Is Raining, Rain (day) , Rain Rate and Rain Storm. 

In this screenshot that I did at 18:05 local time, the last rain was at 18:00 which is correct however todays rain is showing 0 and the correct value is 0.4mm. The Month and Yearly totals are correct and they include the 0.4mm today. I have it setup for 0.2mm settings.

![image](https://github.com/MarcoGos/davis_vantage/assets/22878629/9e7596f6-9f6c-49ec-ad04-e980fc62352a)

I'm not sure if it's related, I'm seeing a lot of these errors now:

````
Logger: pyvpdriver
Source: custom_components/davis_vantage/client.py:84
integration: Davis Vantage ([documentation](https://github.com/MarcoGos/davis_vantage), [issues](https://github.com/MarcoGos/davis_vantage/issues))
First occurred: 5:58:23 PM (15 occurrences)
Last logged: 6:12:28 PM

Check ACK: BAD ('\n\r' != b'\xe0\x06')
Check ACK: BAD ('\n\r' != b'\xe3\x06')
Check ACK: BAD ('\n\r' != '\x06L')
Check ACK: BAD ('\n\r' != b'\xe6\x06')
Check ACK: BAD ('\n\r' != b'\xe9\x06')

**Comment by MarcoGos** [comment]
Check the "Display precision" in the settings of the entity. It probably shows "Default (0)". You can change that to "0.0" to display the one digit precision.

![Scherm­afbeelding 2024-03-07 om 08 44 01](https://github.com/MarcoGos/davis_vantage/assets/44016861/13fc059f-eac5-49bf-a0c2-f6f7e3e163eb)

**Comment by vk2him** [comment]
Thanks @MarcoGos - yes that was my issue - many thanks

**Comment by vk2him** [comment]
@MarcoGos - The Rain rate doesn't seem to be working - it's fixed on 0.0 even though it's been raining here for a while. I also assumed the Rain Rate time was the last time rain was detected (It's called "Time of last rain"? Can that time be added please ?Many thanks for adding this

![image](https://github.com/MarcoGos/davis_vantage/assets/22878629/2e76fb7b-86ef-4cde-9af3-06f647e241e9)

**Comment by MarcoGos** [comment]
Altered the readme and removed the Vue reference because it's not sure this device works.
